app/controllers/api/v0/service_plans_controller.rb

```diff
@@ -19,19 +21,36 @@ def order
 
       private
 
-      def order_service_plan_and_update_task(task, service_plan)
-        task.context = service_plan.order(order_params.slice(:service_parameters, :provider_control_parameters))
-        task.status = "completed"
-        task.save!
+      def service_plan_id
+        params[:service_plan_id].to_i
       end
 
-      def order_params
+      def params_for_order
```

Comment:
Should this cache the hash to prevent generating multiple hashes (and warnings)

Reply:
Sure, that's simple enough.